gguf-py/gguf/gguf_writer.py

```diff
@@ -236,6 +236,8 @@ def write_kv_data_to_file(self) -> None:
             kv_bytes = bytearray()
 
             for key, val in kv_data.items():
+                if key == "general.license" and isinstance(val.value, list):
+                    val.value = ",".join(val.value)
```

Comment:
The approach of joining the license names with commas is good, but this shouldn't be done at the time of serialization, but at the time of when `GGUFWriter.add_license` is called.

So this should be moved either in the `add_license` method of GGUFWriter

https://github.com/ggerganov/llama.cpp/blob/96776405a17034dcfd53d3ddf5d142d34bdbb657/gguf-py/gguf/gguf_writer.py#L523-L524

Or before its passed to `add_license`, which is done in `gguf-py/gguf/metadata.py`

https://github.com/ggerganov/llama.cpp/blob/96776405a17034dcfd53d3ddf5d142d34bdbb657/gguf-py/gguf/metadata.py#L460-L461